### PR TITLE
fix: add default schedule while creating org member

### LINKED ISSUE #16853: [CAL-4423] Adding an invited(autoaccepted) member of a team who hasn't completed onboarding to a team event breaks the booking page
Steps:
- Have a working team event in an organization
- Invite a member to the team. Member gets automatically accepted as it has organization email
- Now add that user to the team event as fixed host
- Go to the booking page for the team event and notice it doesn't show any availability.

By completing the onboarding, we add a schedule for the user after which it starts working.

<sub>[CAL-4423](https://linear.app/calcom/issue/CAL-4423/adding-an-invitedautoaccepted-member-of-a-team-who-hasnt-completed)</sub>

### PR DESCRIPTION
**Pull request by vijayraghav-io**
## What does this PR do?

- Fixes #16853 
- Fixes CAL-4423

BeforeChanges :
https://www.loom.com/share/232fd6ddfdc8407f93d1a967fe6c191f?sid=c9da5ff0-1f45-4d2a-b8fc-e8053a557be1

AfterChanges :
https://www.loom.com/share/26e808cb4d6b4d8abd4288ed096bfd14?sid=514088a9-9244-442f-8265-cca21a23bf38

## Mandatory Tasks (DO NOT REMOVE)

- [x] I have self-reviewed the code (A decent size PR without self-review might be rejected).
- [ ] I have added a Docs issue [here](https://github.com/calcom/docs/issues/new) if this PR makes changes that would require a [documentation change](https://docs.cal.com/docs). If N/A, write N/A here and check the checkbox.
- [ ] I confirm automated tests are in place that prove my fix is effective or that my feature works.

### CODE REVIEW

File: packages/trpc/server/routers/viewer/teams/inviteMember/utils.ts [hunk 1]
```diff
@@ -307,6 +310,8 @@ export async function createNewUsersConnectToOrgIfExists({
 
         const isBecomingAnOrgMember = parentId || isOrg;
 
+        const availability = getAvailabilityFromSchedule(DEFAULT_SCHEDULE);
```
Comment (Ryukemeister) [inline packages/trpc/server/routers/viewer/teams/inviteMember/utils.ts:313]: How about we rename this to `defaultScheduleAvailability`?

File: packages/trpc/server/routers/viewer/teams/inviteMember/utils.ts [hunk 2]
```diff
@@ -340,6 +345,21 @@ export async function createNewUsersConnectToOrgIfExists({
                 accepted: autoAccept, // If the user is invited to a child team, they are automatically accepted
               },
             },
+            // Default schedule
```
Comment (Ryukemeister) [inline packages/trpc/server/routers/viewer/teams/inviteMember/utils.ts:348]: I think its alright if you remove the comment

**Comment by vijayraghav-io** [comment]
Gentle reminder!

**Comment by vijayraghav-io** [comment]
added an e2e test

**Review by anikdhabal — APPROVED** [review]
LGTM!!

**Comment by supalarry** [comment]
@anikdhabal @vijayraghav-io the created schedules are missing timeZone. Anik, hasn't this caused any issues? Or schedules without timeZone default to Europe/London ?

**Comment by anikdhabal** [comment]
Hey @supalarry, I don't think it causes issues. We've done it in other places as well, e.g., in the userRepository. What issue are you anticipating? But yes, adding the timezone would make sense here. @vijayraghav-io, could you check it and do a follow-up?

**Comment by vijayraghav-io** [comment]
> Hey @supalarry, I don't think it causes issues. We've done it in other places as well, e.g., in the userRepository. What issue are you anticipating? But yes, adding the timezone would make sense here. @vijayraghav-io, could you check it and do a follow-up?

@anikdhabal, sure will check this and do a follow-up.

**Comment by vijayraghav-io** [comment]
Hi @supalarry, @anikdhabal,  i checked and verified this , even if timezone is not explicitly input, the schedule is created and availability shown as per default timezone - `Europe/London`.

Updated to create schedule as per input timezone, if timezone is input for `createNewUsersConnectToOrgIfExists` else timezone is saved with default value - `Europe/London`. 

https://github.com/calcom/cal.com/pull/18288

**Pull request merged**